internal/router/router.go

```diff
@@ -0,0 +1,95 @@
+package router
+
+import (
+	"fmt"
+	"github.com/fabric8-services/fabric8-jenkins-idler/internal/api"
+	"github.com/julienschmidt/httprouter"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"net/http"
+	"time"
+)
+
+var routerLogger = log.WithFields(log.Fields{"component": "router"})
+
+const (
+	defaultHttpServerPort = 8080
+	shutdownTimeout       = 5
+)
+
+// Router implements an HTTP server, exposing the REST API of the Idler
+type Router struct {
+	port int
+	srv  *http.Server
+}
+
+// NewRouter creates a new HTTP router for the Idler on the default port.
+func NewRouter(api api.IdlerAPI) *Router {
```

Comment:
Can NewRouter just call NewRouterWithPort and provide default port? Rather than have 2 copies of the code?

Reply:
Of course. Good catch.

Reply:
Fixed